File: ios/FreelanceKar/Views/Screens/ExploreScreen.swift
import SwiftUI

struct ExploreScreen: View {
    @EnvironmentObject var store: AppStore
    private let howItWorks = ["Describe work", "Compare quotes", "Pay them directly"]

    var body: some View {
        let s = store.state
        let city = Selectors.cityOf(s)
        let recents = [
            (label: "Reel editor under ₹1,500", meta: "\(city) · within 10 km", count: "28"),
            (label: "Product photographer", meta: "\(city) · available this week", count: "64"),
        ]

        ScreenShell {
            VStack(alignment: .leading, spacing: 0) {
                HStack {
                    Text("Explore").font(.fkH1).foregroundStyle(FK.text)
                    Spacer()
                    LocationPill(compact: true)
                }
                .padding(.horizontal, 20)
                .padding(.top, 18)

                HStack(spacing: 10) {
                    Image(systemName: "magnifyingglass").font(.system(size: 14)).foregroundStyle(FK.text45)
                    TextField("", text: Binding(get: { s.query }, set: { store.setQuery($0) }),
                              prompt: Text("Search skills, services or people").foregroundStyle(FK.text42))
                        .font(.system(size: 14))
                        .foregroundStyle(FK.text)
                        .onSubmit { if !s.query.trimmingCharacters(in: .whitespaces).isEmpty { store.search(s.query.trimmingCharacters(in: .whitespaces)) } }
                    if !s.query.trimmingCharacters(in: .whitespaces).isEmpty {
                        Button { store.runSearch() } label: {
                            Text("Go").font(.system(size: 11, weight: .semibold)).foregroundStyle(FK.onLime)
                                .padding(.horizontal, 11).padding(.vertical, 7)
                        }
                        .background(FK.lime)
                        .clipShape(RoundedRectangle(cornerRadius: 8, style: .continuous))
                    }
                }
                .padding(.horizontal, 13)
                .padding(.vertical, 11)
                .background(FK.card)
                .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
                .overlay(RoundedRectangle(cornerRadius: 14).stroke(FK.line09, lineWidth: 1))
                .padding(.horizontal, 20)
                .padding(.top, 18)

                (Text("Tell us the work.").foregroundStyle(FK.text) + Text(" We’ll find the person.").foregroundStyle(FK.text40))
                    .font(.fkHero)
                    .padding(.horizontal, 20)
                    .padding(.top, 26)

                VStack(alignment: .leading, spacing: 8) {
                    FKTextArea(placeholder: "Need 8 Instagram reels a month for my café in Dharampeth…",
                               text: Binding(get: { s.brief }, set: { store.setBrief($0) }), minHeight: 62)
                    HStack {
                        HStack(spacing: 11) {
                            Text("◉"); Text("▤"); Text("₹")
                        }.font(.system(size: 14)).foregroundStyle(FK.text45)
                        Spacer()
                        InlineButton(title: "Find people") { store.runSearch() }
                    }
                }
                .padding(14)
                .background(FK.card)
                .clipShape(RoundedRectangle(cornerRadius: 18, style: .continuous))
                .overlay(RoundedRectangle(cornerRadius: 18).stroke(FK.line10, lineWidth: 1))
                .padding(.horizontal, 20)
                .padding(.top, 18)

                FlowChips(items: Seed.suggestionChips) { store.search($0) }
                    .padding(.horizontal, 20)
                    .padding(.top, 16)

                VStack(alignment: .leading, spacing: 11) {
                    Text("Recent searches").font(.fkH4)
                    ForEach(recents, id: \.label) { r in
                        Button { store.search(r.label) } label: {
                            HStack(spacing: 12) {
                                Text("↺").font(.system(size: 13)).foregroundStyle(FK.text40)
                                VStack(alignment: .leading, spacing: 2) {
                                    Text(r.label).font(.system(size: 13, weight: .semibold))
                                    Text(r.meta).font(.system(size: 10.5)).foregroundStyle(FK.text45)
                                }
                                Spacer()
                                Text(r.count).font(.fkMono).foregroundStyle(FK.text40)
                            }
                            .padding(.horizontal, 14)
                            .padding(.vertical, 12)
                        }
                        .foregroundStyle(FK.text)
                        .background(FK.card)
                        .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
                        .overlay(RoundedRectangle(cornerRadius: 14).stroke(FK.line08, lineWidth: 1))
                    }
                }
                .padding(.horizontal, 20)
                .padding(.top, 26)

                VStack(alignment: .leading, spacing: 11) {
                    Text("How hiring works").font(.fkH4)
                    HStack(spacing: 9) {
                        ForEach(Array(howItWorks.enumerated()), id: \.offset) { i, label in
                            VStack(spacing: 4) {
                                Text("\(i + 1)").font(.system(size: 17, weight: .semibold))
                                Text(label).font(.system(size: 10.5)).foregroundStyle(FK.text50).multilineTextAlignment(.center)
                            }
                            .frame(maxWidth: .infinity)
                            .padding(.horizontal, 8)
                            .padding(.vertical, 13)
                            .background(FK.card)
                            .clipShape(RoundedRectangle(cornerRadius: 14, style: .continuous))
                            .overlay(RoundedRectangle(cornerRadius: 14).stroke(FK.line08, lineWidth: 1))
                        }
                    }
                }
                .padding(.horizontal, 20)
                .padding(.top, 24)
                .padding(.bottom, 26)
            }
        }
    }
}

/// Wrapping chip row. Uses a flow layout so tag-style chips wrap onto
/// multiple lines instead of a fixed grid.
struct FlowChips: View {
    var items: [String]
    var action: (String) -> Void

    var body: some View {
        FKFlowLayout(spacing: 8, lineSpacing: 8) {
            ForEach(items, id: \.self) { item in
                Button { action(item) } label: {
                    Text(item)
                        .font(.system(size: 12, weight: .medium))
                        .foregroundStyle(Color(hex: 0xF2F2F0, opacity: 0.75))
                        .padding(.horizontal, 12).padding(.vertical, 8)
                }
                .background(FK.card)
                .clipShape(Capsule())
                .overlay(Capsule().stroke(FK.line09, lineWidth: 1))
            }
        }
    }
}

/// Minimal flow layout (iOS 16+ Layout protocol) that wraps children left to
/// right, starting a new line when the row would overflow.
struct FKFlowLayout: Layout {
    var spacing: CGFloat = 8
    var lineSpacing: CGFloat = 8

    func sizeThatFits(proposal: ProposedViewSize, subviews: Subviews, cache: inout ()) -> CGSize {
        let maxWidth = proposal.width ?? .infinity
        var x: CGFloat = 0
        var y: CGFloat = 0
        var lineHeight: CGFloat = 0
        for view in subviews {
            let size = view.sizeThatFits(.unspecified)
            if x + size.width > maxWidth, x > 0 {
                x = 0
                y += lineHeight + lineSpacing
                lineHeight = 0
            }
            x += size.width + spacing
            lineHeight = max(lineHeight, size.height)
        }
        y += lineHeight
        return CGSize(width: maxWidth.isFinite ? maxWidth : x, height: y)
    }

    func placeSubviews(in bounds: CGRect, proposal: ProposedViewSize, subviews: Subviews, cache: inout ()) {
        let maxWidth = bounds.width
        var x: CGFloat = bounds.minX
        var y: CGFloat = bounds.minY
        var lineHeight: CGFloat = 0
        for view in subviews {
            let size = view.sizeThatFits(.unspecified)
            if x + size.width > bounds.minX + maxWidth, x > bounds.minX {
                x = bounds.minX
                y += lineHeight + lineSpacing
                lineHeight = 0
            }
            view.place(at: CGPoint(x: x, y: y), proposal: .unspecified)
            x += size.width + spacing
            lineHeight = max(lineHeight, size.height)
        }
    }
}

File: android/app/src/main/java/com/freelancekar/app/ui/screens/ExploreScreen.kt
package com.freelancekar.app.ui.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.text.KeyboardActions
import androidx.compose.foundation.text.KeyboardOptions
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.material3.TextField
import androidx.compose.material3.TextFieldDefaults
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.input.ImeAction
import androidx.compose.ui.text.input.KeyboardType
import androidx.compose.ui.unit.dp
import com.freelancekar.app.data.*
import com.freelancekar.app.ui.AppViewModel
import com.freelancekar.app.ui.components.InlineButton
import com.freelancekar.app.ui.components.LocationPill
import com.freelancekar.app.ui.theme.FkColor
import com.freelancekar.app.ui.theme.FkType

private val HOW_IT_WORKS = listOf("Describe work", "Compare quotes", "Pay them directly")

@Composable
fun ExploreScreen(state: AppState, vm: AppViewModel) {
    val city = cityOf(state)
    val recents = listOf(
        Triple("Reel editor under ₹1,500", "$city · within 10 km", "28"),
        Triple("Product photographer", "$city · available this week", "64"),
    )

    Column(Modifier.fillMaxSize().verticalScroll(rememberScrollState())) {
        Row(
            Modifier.fillMaxWidth().padding(horizontal = 20.dp).padding(top = 18.dp),
            horizontalArrangement = Arrangement.SpaceBetween,
        ) {
            Text("Explore", style = FkType.h1)
            LocationPill(state, { vm.openCity() }, compact = true)
        }

        Row(
            Modifier
                .fillMaxWidth()
                .padding(horizontal = 20.dp)
                .padding(top = 18.dp)
                .clip(RoundedCornerShape(14.dp))
                .background(FkColor.card)
                .border(1.dp, FkColor.line09, RoundedCornerShape(14.dp))
                .padding(horizontal = 13.dp, vertical = 4.dp),
        ) {
            Text("⌕", style = FkType.body.copy(color = FkColor.text45), modifier = Modifier.padding(top = 12.dp, end = 8.dp))
            TextField(
                value = state.query,
                onValueChange = { vm.setQuery(it) },
                placeholder = { Text("Search skills, services or people", style = FkType.body.copy(color = FkColor.text42)) },
                modifier = Modifier.weight(1f),
                singleLine = true,
                colors = TextFieldDefaults.colors(
                    focusedContainerColor = FkColor.card, unfocusedContainerColor = FkColor.card,
                    focusedIndicatorColor = androidx.compose.ui.graphics.Color.Transparent,
                    unfocusedIndicatorColor = androidx.compose.ui.graphics.Color.Transparent,
                ),
                textStyle = FkType.body.copy(color = FkColor.text),
                keyboardOptions = KeyboardOptions(imeAction = ImeAction.Search),
                keyboardActions = KeyboardActions(onSearch = {
                    if (state.query.trim().isNotEmpty()) vm.searchPublic(state.query.trim())
                }),
            )
            if (state.query.trim().isNotEmpty()) {
                Box(Modifier.padding(vertical = 8.dp)) {
                    InlineButton({ vm.runSearch() }, text = "Go")
                }
            }
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 26.dp)) {
            Text("Tell us the work. We'll find the person.", style = FkType.hero)
        }

        Column(
            Modifier
                .padding(horizontal = 20.dp)
                .padding(top = 18.dp)
                .clip(RoundedCornerShape(18.dp))
                .background(FkColor.card)
                .border(1.dp, FkColor.line10, RoundedCornerShape(18.dp))
                .padding(14.dp),
        ) {
            TextField(
                value = state.brief,
                onValueChange = { vm.setBrief(it) },
                placeholder = { Text("Need 8 Instagram reels a month for my café in Dharampeth…", style = FkType.body.copy(color = FkColor.text42)) },
                modifier = Modifier.fillMaxWidth().heightIn(min = 62.dp),
                colors = TextFieldDefaults.colors(
                    focusedContainerColor = FkColor.card, unfocusedContainerColor = FkColor.card,
                    focusedIndicatorColor = androidx.compose.ui.graphics.Color.Transparent,
                    unfocusedIndicatorColor = androidx.compose.ui.graphics.Color.Transparent,
                ),
                textStyle = FkType.body.copy(color = FkColor.text),
            )
            Row(Modifier.fillMaxWidth().padding(top = 8.dp), horizontalArrangement = Arrangement.SpaceBetween) {
                Row {
                    Text("◉  ▤  ₹", style = FkType.body.copy(color = FkColor.text45))
                }
                InlineButton({ vm.runSearch() }, text = "Find people")
            }
        }

        androidx.compose.foundation.layout.FlowRow(
            modifier = Modifier.padding(horizontal = 20.dp).padding(top = 16.dp),
            horizontalArrangement = Arrangement.spacedBy(8.dp),
        ) {
            SUGGESTION_CHIPS.forEach { name ->
                Box(
                    Modifier
                        .clip(CircleShape)
                        .background(FkColor.card)
                        .border(1.dp, FkColor.line09, CircleShape)
                        .clickable { vm.searchPublic(name) }
                        .padding(horizontal = 12.dp, vertical = 8.dp),
                ) {
                    Text(name, style = FkType.meta.copy(color = FkColor.text.copy(alpha = 0.75f)))
                }
            }
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 26.dp)) {
            Text("Recent searches", style = FkType.h4)
            Spacer(Modifier.height(11.dp))
            Column(verticalArrangement = Arrangement.spacedBy(9.dp)) {
                recents.forEach { (label, meta, count) ->
                    Row(
                        Modifier
                            .fillMaxWidth()
                            .clip(RoundedCornerShape(14.dp))
                            .background(FkColor.card)
                            .border(1.dp, FkColor.line08, RoundedCornerShape(14.dp))
                            .clickable { vm.searchPublic(label) }
                            .padding(horizontal = 14.dp, vertical = 12.dp),
                        verticalAlignment = androidx.compose.ui.Alignment.CenterVertically,
                    ) {
                        Text("↺", style = FkType.meta.copy(color = FkColor.text40))
                        Spacer(Modifier.width(10.dp))
                        Column(Modifier.weight(1f), verticalArrangement = Arrangement.spacedBy(2.dp)) {
                            Text(label, style = FkType.cardTitle)
                            Text(meta, style = FkType.metaXs.copy(color = FkColor.text45))
                        }
                        Text(count, style = FkType.mono.copy(color = FkColor.text40))
                    }
                }
            }
        }

        Column(Modifier.padding(horizontal = 20.dp).padding(top = 24.dp, bottom = 26.dp)) {
            Text("How hiring works", style = FkType.h4)
            Spacer(Modifier.height(11.dp))
            Row(horizontalArrangement = Arrangement.spacedBy(9.dp)) {
                HOW_IT_WORKS.forEachIndexed { i, label ->
                    Column(
                        Modifier
                            .weight(1f)
                            .clip(RoundedCornerShape(14.dp))
                            .background(FkColor.card)
                            .border(1.dp, FkColor.line08, RoundedCornerShape(14.dp))
                            .padding(horizontal = 8.dp, vertical = 13.dp),
                        horizontalAlignment = androidx.compose.ui.Alignment.CenterHorizontally,
                        verticalArrangement = Arrangement.spacedBy(4.dp),
                    ) {
                        Text("${i + 1}", style = FkType.h4)
                        Text(label, style = FkType.metaXs.copy(color = FkColor.text50), textAlign = androidx.compose.ui.text.style.TextAlign.Center)
                    }
                }
            }
        }
    }
}
